phoenix-read-write/src/main/java/com/cloudera/cod/examples/sql/AbstractSqlClient.java

```diff
@@ -95,4 +100,144 @@ public void runWrites(String tableName, Connection conn, int numRecords) throws
+    connection.commit();
+  }
+
+  /**
+   *
+   * @param conn connection used for performing transactions.
+   * @param commit whether to commit the order transaction.
+   * @param orderId Order ID of the Creating Order.
+   * @param customerId Customer ID of the customer made an order.
+   * @param itemVsQuantity Items selected with quantities for order.
+   * @throws SQLException
+   */
+  public void createOrder(Connection conn, boolean commit, String orderId, String customerId,
+      Map<String, Integer> itemVsQuantity) throws SQLException {
+    conn.setAutoCommit(false);
+    Statement stmt = conn.createStatement();
+    stmt.execute("UPSERT INTO \"ORDER\" VALUES('" + orderId + "','" + customerId + "',"
+        + " CURRENT_DATE()||' '|| CURRENT_TIME())");
+    for(Entry<String, Integer> item: itemVsQuantity.entrySet()) {
+      String itemID = item.getKey();
+      int saleQuantity = item.getValue();
+      stmt.execute("UPSERT INTO ORDER_LINE_ITEM VALUES('"+ orderId+"','" +itemID+"',1)");
+      stmt.execute("UPSERT INTO ITEM(ID, QUANTITY)"
+          + " SELECT '"+itemID+"', QUANTITY - " + saleQuantity + " FROM ITEM "
+          + " WHERE ID = '" + itemID + "'");
+    }
+    if(commit) {
+      conn.commit();
+    }
+  }
+
+  public void runTransactions(String jdbcUrl) throws SQLException {
+    Connection conn = DriverManager.getConnection(jdbcUrl);
+    createTransactionalTables(conn);
+    conn.setAutoCommit(false);
+    populateData(conn);
+    // Create Multiple Connections for different terminal clients to make order entries.
+    Connection firstTerminalConn = DriverManager.getConnection(jdbcUrl);
+    Connection secondTerminalConn = DriverManager.getConnection(jdbcUrl);
+    Connection thirdTerminalConn = DriverManager.getConnection(jdbcUrl);
+
+    // Create Order OR001 with Items Book, Pen, Soap, Shampoo for customer John at first terminal
+    // and committed.
+    Map<String, Integer> firstTerminalItems = new HashMap<>();
+    firstTerminalItems.put("ITM001", 2);
+    firstTerminalItems.put("ITM002", 3);
+    firstTerminalItems.put("ITM003", 2);
+    createOrder(firstTerminalConn, true, "OR001", "CU001",
+        firstTerminalItems);
+
+    // Create Order OR002 with Items Phone, Charger for customer Angel at second terminal but
+    // not committed.
+    Map<String, Integer> secondTerminalItems = new HashMap<>();
+    secondTerminalItems.put("ITM005", 1);
+    secondTerminalItems.put("ITM006", 1);
+    createOrder(secondTerminalConn, false, "OR002", "CU002",
+        secondTerminalItems);
+
+    // Create Order OR003 with Items Book, Pen, Phone, Charger for customer David at third terminal
+    // and committed
+    Map<String, Integer> thirdTerminalItems = new HashMap<>();
+    thirdTerminalItems.put("ITM001", 1);
+    thirdTerminalItems.put("ITM002", 1);
+    thirdTerminalItems.put("ITM005", 1);
+    thirdTerminalItems.put("ITM006", 1);
+    createOrder(thirdTerminalConn, true, "OR003", "CU003",
```

Comment:
Calling  commit here explicitly, instead of passing it as a parameter would improve readability.

Reply:
Done.